redpanda/templates/service.loadbalancer.yaml

```diff
@@ -0,0 +1,54 @@
+{{/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/}}
+{{- if .Values.loadBalancer.enabled }}
+{{- range untilStep 0 (.Values.statefulset.replicas|int) 1 }}
+# This service is to expose a pod in the statefulset with a loadbalancer.
+# Includes integration with https://github.com/kubernetes-sigs/external-dns
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "redpanda.fullname" $ }}-{{ . }}
+  namespace: {{ $.Release.Namespace | quote }}
+  labels:
+    helm.sh/chart: {{ template "redpanda.chart" $ }}
+    app.kubernetes.io/name: {{ template "redpanda.name" $ }}
+    app.kubernetes.io/instance: {{ $.Release.Name | quote }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service | quote }}
+    app.kubernetes.io/component: {{ template "redpanda.name" $ }}
+  {{- with $.Values.labels }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: {{ template "redpanda.fullname" $ }}-{{ . }}.{{$.Values.loadBalancer.parentZone}}
```

Comment:
There could be problem when I create 2 releases with the same name and 2 different namespaces there will be clash of the hostnames.

Reply:
I think that's expected, though. The answer is don't do that.

```
helm install  --set loadBalancer.parentZone=unique.example.com
```

Reply:
Maybe the default should include the namespace to make that clearer, and then a comment saying reaffirming the unique requirement?

Reply:
Agree. Adding namespace it will help to create unique hostnames in on cluster.